Forza-Mods-AIO/Forza-Mods-AIO/ViewModels/Windows/MainWindowViewModel.cs

```diff
@@ -108,10 +113,74 @@ private void InitializeViewModel()
 
         CurrentView = Resources.Pages.GetPage(typeof(AioInfo));
 
-        SetupAttach();
+        //SetupAttach();
         _isInitialized = true;
+        Task task = CheckForUpdates();
     }
-    
+
+    // update check
+
+    private string GitHubRepoUrl = "https://api.github.com/repos/ForzaMods/Forza-Mods-AIO/releases/latest";
+    private string GitUpdate = "https://github.com/ForzaMods/Forza-Mods-AIO/releases";
+    private string? latestGit;
+
+    public async Task CheckGit()
+    {
+        using (var httpClient = new HttpClient())
+        {
+            httpClient.DefaultRequestHeaders.UserAgent.ParseAdd("Opera/9.00 (Nintendo Wii; 1309-9)");
+
+            try
+            {
+                HttpResponseMessage response = await httpClient.GetAsync(GitHubRepoUrl);
+
+                if (response.IsSuccessStatusCode)
+                {
+                    string json = await response.Content.ReadAsStringAsync();
+                    var release = JsonSerializer.Deserialize<GithubRelease>(json);
+                    latestGit = release?.tag_name;
+                }
+            }
+            catch (HttpRequestException ex)
+            {
+                MessageBox.Show($"Error fetching latest release: {ex.Message}", "Exception", MessageBoxButton.OK, MessageBoxImage.Information);
+            }
+        }
+    }
+    private void CompareVer()
+    {
+        Version? assemblyVersion = Assembly.GetExecutingAssembly().GetName().Version;
+        string curVersion = assemblyVersion?.ToString() ?? "Unknown";
+
+        if (latestGit != null && assemblyVersion != null)
+        {
+            if (Version.Parse(latestGit) > assemblyVersion)
+            {
+                MessageBox.Show($"Update to version {latestGit}", "Update Available", MessageBoxButton.OK, MessageBoxImage.Warning);

```

Comment:
can you make this a yes/no prompt and if user chooses no, then just don't update?

Reply:
Updated.
[](url)https://github.com/ForzaMods/Forza-Mods-AIO/pull/41/commits/bf059f827abf61ce03a8c4c6779c5df4c7283637